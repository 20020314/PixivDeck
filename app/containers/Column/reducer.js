// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
