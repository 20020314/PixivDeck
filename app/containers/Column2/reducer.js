// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'

type ColumnId = string
type BodyId = string
type BodyType = 'RANKING' | 'BOOKMARK'

export type Column = {
  id: ColumnId,
  title: string,
  bodyId: BodyId,
  type: BodyType,
}

export type Body = {
  id: BodyId,
  type: BodyType,
}

export type Ranking = {
  endpoint: '/v1/illust/ranking',
  params: {
    mode: string,
    offset: number,
  },
} & Body

export type State = {
  [id: string]: Column,
}

const initialState: State = {}

export default function(
  state: State = initialState,
  action: Action
): $Shape<State> {
  switch (action.type) {
    case Actions.ADD_COLUMN:
      return { ...state, [action.id]: action.body }
    default:
      return state
  }
}
