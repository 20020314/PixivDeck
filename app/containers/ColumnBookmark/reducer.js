// @flow
import { handleRehydrate } from 'util/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Restrict = 'public' | 'private'

export type Endpoint = '/v1/user/bookmarks/illust'

export type ColumnId = Restrict

export type ColumnBookmark = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [ColumnId]: ColumnBookmark }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_BOOKMARK_COLUMN_SUCCESS:
      return { ...state, [action.id]: { illustIds: [], nextUrl: null } }

    case Actions.SET_NEXT_URL: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], nextUrl: action.nextUrl },
      }
    }

    case Actions.FETCH_BOOKMARK_SUCCESS:
    case Actions.FETCH_NEXT_BOOKMARK_SUCCESS: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], illustIds: action.ids },
      }
    }

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnBookmark')
    default:
      return state
  }
}
