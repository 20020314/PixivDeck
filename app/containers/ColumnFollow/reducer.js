// @flow
import update from 'utils/update'
import { handleRehydrate } from 'utils/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Restrict = 'public' | 'private'

export type Endpoint = '/v1/user/follows/illust'

export type ColumnId = Restrict

export type ColumnFollow = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [ColumnId]: ColumnFollow }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_FOLLOW_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    case Actions.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })

    case Actions.FETCH_FOLLOW_SUCCESS:
    case Actions.FETCH_NEXT_FOLLOW_SUCCESS:
      return update(state, action, { illustIds: action.ids })

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnFollow')
    default:
      return state
  }
}
