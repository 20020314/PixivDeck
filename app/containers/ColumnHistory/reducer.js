// @flow
import { union } from 'lodash'
import { REHYDRATE } from 'redux-persist/constants'
import type { Action } from './actionTypes'
import * as Actions from './constants'

export type ColumnId = 'RANKING'

export type State = {
  illustIds: Array<number>,
}

const initialState: State = {
  illustIds: [],
}

export default function(
  state: State = initialState,
  action: Action
): $Shape<State> {
  switch (action.type) {
    case Actions.ADD_HISTORY:
      return { illustIds: union([action.id, ...state.illustIds]) }
    case REHYDRATE: {
      // $FlowFixMe
      const oldState: State = action.payload.ColumnHistory
      if (oldState) {
        return { illustIds: [] }
      }
      return state
    }
    default:
      return state
  }
}
