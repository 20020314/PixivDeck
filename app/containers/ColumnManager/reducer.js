// @flow
import update from 'util/update'
import type { Action } from './actionTypes'
import * as Actions from './constants'

export type ColumnManagerId = string
export type ColumnId = string
export type ColumnType =
  | 'RANKING'
  | 'RANKING_R18'
  | 'BOOKMARK'
  | 'FOLLOW'
  | 'USER_ILLUST'
  | 'SEARCH'

export type ColumnManager = {
  columnId: ColumnId,
  type: ColumnType,
}

export type Column = {
  title: string,
}

export type State = {
  [id: ColumnManagerId]: ColumnManager,
}

export default function(state: State = {}, action: Action): $Shape<State> {
  switch (action.type) {
    case Actions.ADD_COLUMN:
      return update(state, action, action.maneger)
    default:
      return state
  }
}
