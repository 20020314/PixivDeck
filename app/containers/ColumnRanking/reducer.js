// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Mode =
  | 'day'
  | 'week'
  | 'month'
  | 'day_male'
  | 'day_female'
  | 'week_original'
  | 'week_rookie'

export type Endpoint = '/v1/illust/ranking'

export type ColumnId = Mode

export type ColumnRanking = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [Mode]: ColumnRanking }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_RANKING_COLUMN_SUCCESS:
      return { ...state, [action.id]: { illustIds: [], nextUrl: null } }

    case Actions.SET_NEXT_URL: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], nextUrl: action.nextUrl },
      }
    }

    case Actions.FETCH_RANKING_SUCCESS:
    case Actions.FETCH_NEXT_RANKING_SUCCESS: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], illustIds: action.ids },
      }
    }

    // TODO REHYDRATEをまとめる
    case REHYDRATE: {
      // $FlowFixMe
      const oldState = action.payload.ColumnRanking

      if (oldState) {
        const newState = Object.keys(oldState).reduce((acc, key) => {
          acc[key] = {
            illustIds: [],
            nextUrl: null,
          }
          return acc
        }, {})
        return newState
      }
      return state
    }

    default:
      return state
  }
}
