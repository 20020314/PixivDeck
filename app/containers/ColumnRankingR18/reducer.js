// @flow
import { handleRehydrate } from 'util/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type R18Mode =
  | 'day_r18'
  | 'week_r18'
  | 'day_male_r18'
  | 'day_female_r18'
  | 'week_r18g'

export type Endpoint = '/v1/illust/ranking'

export type ColumnId = R18Mode

export type ColumnRanking = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [R18Mode]: ColumnRanking }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_RANKING_R18_COLUMN_SUCCESS:
      return { ...state, [action.id]: { illustIds: [], nextUrl: null } }

    case Actions.SET_NEXT_URL: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], nextUrl: action.nextUrl },
      }
    }

    case Actions.FETCH_NEXT_RANKING_R18_SUCCESS:
    case Actions.FETCH_RANKING_R18_SUCCESS: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], illustIds: action.ids },
      }
    }

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnRankingR18')

    default:
      return state
  }
}
