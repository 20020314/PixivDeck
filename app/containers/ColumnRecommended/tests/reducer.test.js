// @flow
import reducer from '../reducer'

test('default action', () => {
  // $FlowFixMe
  expect(reducer({}, { type: 'default action' })).toMatchSnapshot()
})
