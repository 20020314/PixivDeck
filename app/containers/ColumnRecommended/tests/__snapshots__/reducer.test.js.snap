// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default action 1`] = `Object {}`;
